garak/cli.py

```diff
@@ -429,6 +431,60 @@ def main(arguments=None) -> None:
             command.list_config()
 
+        elif args.fix:
+            from garak.resources import fixer
+            import json
+            import yaml
+
+            # process all possible configuration entries
+            # should this restrict the config updates to a single fixable value?
+            # for example allowed commands:
+            # --fix --config filename.yaml
+            # --fix --generator_option_file filename.json
+            # --fix --generator_options json
+            #
+            # disallowed commands:
+            # --fix --config filename.yaml --generator_option_file filename.json
+            # --fix --generator_option_file filename.json --probe_option_file filename.json
+            #
+            # already unsupported as only one is held:
+            # --fix --generator_option_file filename.json --generator_options json_data
+            #
+            # How should this handle garak.site.yaml? Only if --fix was provided and no other options offered?
+            # For now process all files registered a part of the config
+            has_changes = False
+            if has_config_file_or_json:
+                for plugin_type, plugin_plural in plugin_types:
+                    # cli plugins options stub out only a "plugins" sub key
+                    plugin_cli_config = parse_cli_plugin_config(plugin_type, args)
+                    if plugin_cli_config is not None:
+                        cli_config = {
+                            "plugins": {f"{plugin_plural}": plugin_cli_config}
+                        }
+                        migrated_config = fixer.migrate(cli_config)
+                        if cli_config != migrated_config:
+                            has_changes = True
+                            msg = f"Updated '{plugin_type}' configuration: \n"
+                            msg += json.dumps(
+                                migrated_config["plugins"][plugin_plural], indent=2
+                            )  # pretty print the config in json
+                            print(msg)
+            else:
+                # check if garak.site.yaml needs to be fixed up?
+                for filename in _config.config_files:
+                    with open(filename, encoding="UTF-8") as file:
+                        cli_config = yaml.safe_load(file)
+                        migrated_config = fixer.migrate(cli_config)
+                        if cli_config != migrated_config:
+                            has_changes = True
+                            msg = f"Updated {filename}: \n"
+                            msg += yaml.dump(migrated_config)
+                            print(msg)
+            # should this add support for --*_spec entries passed on cli?
+            if has_changes:
+                exit(1)  # exit with error code to denote changes
+            else:
+                print("No revisions applied please verify options provided for `--fix`")
```

Comment:
Worth discussing, Can you elaborate on reasons? 

I am currently of the option that this is `config` related the primary role of `cli` code is parsing config. `command` so far, aligns with items that should be possible to call from `interactive` mode. This functionality may more may not be valuable in that context and uses the parsed args array in `parse_cli_plugin_config` a way that suggests this is currently the more appropriate location to maintain encapsulation. A future refactor may extract some config parsing from `cli` and change this view.

Reply:
to my mind:
* `cli` is ideally items only related to processing cli input
* `command` contains encapsulation of functions / activities garak can perform
* orchestration of runs is in a grey zone pending better encapsulation, that will be resolved with an API design, and I prefer `command` for this, seeing as run orchestration subfunctions are here already

following the first point, any items not related solely to command line interface, shouldn't really be in `cli`. this code block contains code that isn't about processing cli input. so, it's better off outside the `cli` module

ideally, i think it should be entirely possible to access all of garak's functionality without using `garak.cli`. so that in effect, `cli` provides only an interface to garak proper. these config updates are not intrinsic to the command line interface, and one would conceivably like to be able to access them via other routes as & when those open up.

Reply:
huh, yeah, i see what you mean - good point. i'm fine with taking the path of least resistance and leaving this as-is, in that case